src/vcr_clj/core.clj

```diff
@@ -159,5 +179,7 @@
                   while recording, which can be useful for doing things
                   like ensuring serializability.
     }"
-  [cname specs & body]
-  `(with-cassette-fn* ~cname ~specs (fn [] ~@body)))
+  [cdata specs & body]
+  `(let [cname# (if (map? ~cdata) (:name ~cdata) ~cdata)
+         opts# (when (map? ~cdata) ~cdata)]
```

Comment:
unquoting `cdata` multiple times like this means the first arg to `with-cassette` gets duplicated in the expansion; the cleaner approach is to add `cdata# ~cdata` as the first clause in the `let` and use `cdata#` everywhere else

however, it might be even better to just pass `cdata` straight through and do this logic in `with-cassette-fn*`; that way they'd have a more consistent interface